### DISCUSSION TITLE
Question on [ComputeLagrangianStrain]

### DISCUSSION THREAD

**Discussion by rh201** [idaholab/moose#25553] [Q&A Modules: Solid mechanics] [answered]
Dear MOOSEers,

in the page of ComputeLagrangianStrain https://mooseframework.inl.gov/moose/source/materials/lagrangian/ComputeLagrangianStrain.html, there is the note reading"The strain calculator does not apply the eigenstrains to the deformation gradient, only the incremental and integrated strain measures." 
So dose it mean for small strain, it's used to calculate eigenstrain from total strain as `\epsilon = \epsilon_mech + \epsilon_eigen`? 
and for large kinematics, the eigen part like the thermal deformation gradient is not excluded from the total deformation gradient, as decomposed `F=F^\theta F^mech`? and what if the thermal part should be excluded?

**Comment by hugary1995** [comment, marked as answer]
> So dose it mean for small strain, it's used to calculate eigenstrain from total strain as \epsilon = \epsilon_mech + \epsilon_eigen?

This is correct. The stress calculator computes stress as function of mechanical strain.

> and for large kinematics, the eigen part like the thermal deformation gradient is not excluded from the total deformation gradient, as decomposed F=F^\theta F^mech? and what if the thermal part should be excluded?

No, for large kinematics the eigenstrains are also accounted for if you use one of the objectively integrated stress calculator, i.e. we advect a small stress-strain model to render it as a large deformation model.

However, if you use some hyperelastic material model and directly compute stress as a function of the deformation gradient, then indeed the eigenstrains are not being excluded. That is a missing capability in the new system.

**Reply by rh201** [reply]
Thanks @hugary1995
